airflow/models.py

```diff
@@ -2485,7 +2485,10 @@ def __lt__(self, other):
     def __hash__(self):
         hash_components = [type(self)]
         for c in self._comps:
-            val = getattr(self, c, None)
+            if c == 'task_ids':
```

Comment:
Nit: Rather than having a special case consider making dag.task_ids a DAG property instead (then getattr would still work for it)